src/ui/Debug/Debug.ts

```diff
@@ -63,6 +63,109 @@ export class Debug extends RootComponent {
     this.collapsedSections = this.localStorageDebug.load() || [];
   }
 
+  public buildFieldsSection(result: IQueryResult) {
+    return this.fetchFields()
+      .then((fieldDescriptions: { [field: string]: IFieldDescription }) => {
```

Comment:
To add typing to JSON structure like this, you can use `IStringMap<IFieldDescription>`, it's a bit more concise.